tools/munin/minion

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env perl
+use strict;
+use warnings;
+use v5.22;
+
+# Usage
+#
+# Add this to /etc/munin/plugin-conf.d/munin-node:
+#
+#   [minion_*]
+#   env.url https://openqa.opensuse.org/admin/influxdb/minion
+#
+# Put script into /usr/lib/munin/plugins/minion_ (or symlink)
+
+# Create 3 symlinks in /etc/munin/plugins:
+# minion_jobs                -> /usr/lib/munin/plugins/minion_
+# minion_workers             -> /usr/lib/munin/plugins/minion_
+# minion_jobs_hook_rc_failed -> /usr/lib/munin/plugins/minion_
+
+my $url = $ENV{url} || "https://openqa.opensuse.org/admin/influxdb/minion";
+
+my %config = (
+    minion_jobs => {
+        states => [qw/ active delayed failed inactive /],
+        title => "Minion Jobs",
+        warning => { failed => 300 },
+        critical => { failed => 400 },
+    },
+    minion_workers => {
+        states => [qw/ active inactive registered /],
+        title => "Minion Workers",
+    },
+    minion_jobs_hook_rc_failed => {
+        states => [qw/ rc_failed_per_10min /],
+        title => "hook failed",
+        warning => { rc_failed_per_10min => 1 },
+        critical => { rc_failed_per_10min => 2 },
+    },
+);
+
+my ($category) = $0 =~ m/(minion_\w+)/;
+my $config = $config{ $category };
+my $states = $config->{states};
+
+if ( defined $ARGV[0] and $ARGV[0] eq "config" ) {
+    config();
+    exit;
+}
+
+#openqa_minion_jobs,url=https://openqa.opensuse.org active=0i,delayed=0i,failed=352i,inactive=0i
+#openqa_minion_jobs_hook_rc_failed,url=https://openqa.opensuse.org rc_failed_per_10min=0i 16861374000000000
+#openqa_minion_workers,url=https://openqa.opensuse.org active=0i,inactive=1i,registered=1i
+my @data = qx{curl -s $url};
+
+my %values;
+for my $line (@data) {
+    my ($metric, $url, $values) = $line =~ m/^(\w+),url=(\S+) (\S+)/;
+    next unless $metric eq "openqa_$category";
+    %values = map { s/i$//r } map { split m/=/, $_ } split m/,/, $values;
```

Comment:
Can we have a simple unit test for this?

Reply:
I added a test